hack/managed-kyma-migration/README.md

```diff
@@ -0,0 +1,109 @@
+
+#### SKR with an existing module
+
+1. Create Dev SKR.
+2. Add the Keda module to the Kyma CR.
+   ```yaml
+   spec:
+     modules:
+       - name: keda
+   ```
+3. Execute the migration.
+4. Verify that `api-gateway-manager` is installed and the APIGateway CR's status is `Ready`.
+5. Verify that `api-gateway` deployment is not present.
+
+## Module's rollout and migration
+
+### Preparations
+
+1. Executing `kcp taskrun` requires the path to the kubeconfig file of the corresponding Gardener project and permissions to list/get shoots.
+
+### Dev
+
+#### Prerequisites
+
+- Reconciliation is disabled for Dev environment (kyma/management-plane-config `PR #4485`)
+
+#### Migration procedure
+
+1. Apply the ModuleTemplate for both `fast` and `regular` channel to DEV Control Plane.
+2. Verify that the ModuleTemplate in `fast` and `regular` channel is available on `Dev` environment SKRs.
+3. `kcp login` to Dev and run the migration script on all SKRs. To do that, you can use the following command:
+   ```shell
+   kcp taskrun --gardener-kubeconfig {PATH TO GARDENER PROJECT KUBECONFIG} --gardener-namespace kyma-dev -t all -- ./managed-kyma-migration.sh
+   ```
+4. Verify that the migration worked as expected by checking the status of APIGateway manifests on Control Plane.
+   ```shell
+   kubectl get manifests -n kcp-system -o custom-columns=NAME:metadata.name,STATE:status.state | grep api-gateway
+   ```
+5. Verify that `api-gateway` deployment is not present on any of the SKRs
+   ```shell
+   kcp taskrun --gardener-kubeconfig {PATH TO GARDENER PROJECT KUBECONFIG} --gardener-namespace kyma-dev -t all -- kubectl get deployment -n kyma-system api-gateway 2>/dev/null
+   ```
+
+### Stage
+
+Perform the rollout to Stage together with the SRE team. Since they have already performed the rollout for other modules, they might suggest a different rollout strategy.
+
+#### Prerequisites
+
+- Reconciliation is disabled for Stage environment (kyma/management-plane-config `PR #4486`)
+
+#### Migration procedure
+
+1. Apply the ModuleTemplate for both `fast` and `regular` channel to Stage Control Plane.
+2. Verify that the ModuleTemplate in `fast` and `regular` channel is available on `Stage` environment SKRs.
+3. `kcp login` to Stage, select some SKRs on `Kyma-Test/Kyma-Integration`, and run `managed-kyma-migration.sh` on them using `kcp taskrun`.
+4. Verify if the migration was successful on the SKRs by checking the status of APIGateway CR and the reconciler's components.
+5. Run `managed-kyma-migration.sh` for all SKRs in Kyma-Test and Kyma-Integration global accounts.
+6. Verify if the migration worked as expected.
+7. Run `managed-kyma-migration.sh` for the whole Canary landscape.
+8. Verify if the migration worked as expected.
+
+### Prod
+
+Perform the rollout to `Prod` together with the SRE team. Since they have already performed the rollout for other modules, they might suggest a different rollout strategy.
```

Comment:
```suggestion
Perform the rollout to Prod together with the SRE team. Since they have already performed the rollout for other modules, they might suggest a different rollout strategy.
```